tests/antithesis/test-template/robustness/main.go

```diff
@@ -149,29 +149,51 @@ func runTraffic(ctx context.Context, lg *zap.Logger, hosts []string, baseTime ti
 
 	finish := closeAfter(ctx, duration)
 	reports := []report.ClientReport{r}
+	watchReport := []report.ClientReport{}
 	var mux sync.Mutex
 	var wg sync.WaitGroup
-	for i := 0; i < profile.ClientCount; i++ {
-		c := connect([]string{hosts[i%len(hosts)]}, ids, baseTime)
-		defer c.Close()
-		wg.Add(1)
-		go func(c *client.RecordingClient) {
-			defer wg.Done()
+	maxRevisionChan := make(chan int64, 1)
+	go func() {
+		defer close(maxRevisionChan)
+		for i := 0; i < profile.ClientCount; i++ {
+			c := connect([]string{hosts[i%len(hosts)]}, ids, baseTime)
 			defer c.Close()
-
-			traffic.EtcdAntithesis.RunTrafficLoop(ctx, c, limiter,
-				ids,
-				storage,
-				concurrencyLimiter,
-				finish,
-			)
-			mux.Lock()
-			reports = append(reports, c.Report())
-			mux.Unlock()
-		}(c)
-	}
-	wg.Wait()
-	return reports
+			wg.Add(1)
+			go func(c *client.RecordingClient) {
+				defer wg.Done()
+				defer c.Close()
+
+				traffic.EtcdAntithesis.RunTrafficLoop(ctx, c, limiter,
+					ids,
+					storage,
+					concurrencyLimiter,
+					finish,
+				)
+				mux.Lock()
+				reports = append(reports, c.Report())
+				mux.Unlock()
+			}(c)
+		}
+		wg.Wait()
+		maxRevision := report.OperationsMaxRevision(reports)
+		maxRevisionChan <- maxRevision
+		lg.Info("Finished simulating Traffic", zap.Int64("max-revision", maxRevision))
+	}()
+	go func() {
+		endpoints := make([][]string, len(hosts))
+		for _, host := range hosts {
+			endpoints = append(endpoints, []string{host})
+		}
+		watchConfig := client.WatchConfig{
+			RequestProgress: true,
+		}
```

Comment:
Move this code outside goroutine. Goroutines should have minimal code needed for coordination between them.